### DISCUSSION TITLE
Pipenv doesn't support newlines in `.env` file

### DISCUSSION THREAD

**Discussion by ekohl** [sanitizers/octomachinery#10] [Q&A] [answered]
Pipenv loads `.env` by default, but it doesn't support newlines. That means `\n` ends up as a literal `\n` (`\\n`). This is problematic because it's used in the private key. The workaround I've found is using `PIPENV_DONT_LOAD_ENV=1` to force disable loading it.

The downside is that the `environ` module doesn't clear the `ENV` variable if it's already set which Pipenv does do by default.

**Comment by webknjaz** [comment]
That's by design. And I didn't mean to support Pipenv anyway as I think it's a failure doing things it's not supposed to do.

**Reply by asjav2** [reply]
This is a piece of great information about all **[Microsoft Software](https://software100.online/)** List with a Low Cost for Computer Users. I am happy to read more all about Microsoft Software List with Low Cost for Computer Uses.

**Comment by webknjaz** [comment]
Docs PRs are welcome anyway.

**Comment by ekohl** [comment, marked as answer]
IMHO it's a docs thing as well. Linking to https://pipenv.readthedocs.io/en/latest/advanced/#automatic-loading-of-env in the tutorial with a warning that pipenv has some issues would have saved me some time.

**Comment by webknjaz** [comment]
This should be addressed by https://github.com/sanitizers/octomachinery/commit/f3d4b329b5e8c83d5ed92b42d3042a54a1e9b3f1.
And to change the tutorial contents you can send PRs against https://github.com/sanitizers/github-bot-tutorial.
